test/integration/targets/ansible-galaxy-collection/library/reset_pulp.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2020, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: reset_pulp
+short_description: Resets pulp back to the initial state
+description:
+- See short_description
+options:
+  pulp_api:
+    description:
+    - The Pulp API endpoint.
+    required: yes
+    type: str
+  galaxy_ng_server:
+    description:
+    - The Galaxy NG API endpoint.
+    required: yes
+    type: str
+  url_username:
+    description:
+    - The username to use when authenticating against Pulp.
+    required: yes
+    type: str
+  url_password:
+    description:
+    - The password to use when authenticating against Pulp.
+    required: yes
+    type: str
+  repository:
+    description:
+    - The name of the repository to create.
+    - This should match C(GALAXY_API_DEFAULT_DISTRIBUTION_BASE_PATH) in C(/etc/pulp/settings.py) or use the default of
+      C(published).
+    required: yes
+    type: str
+  namespaces:
+    description:
+    - A list of namespaces to create.
+    required: yes
+    type: list
+    elements: str
+author:
+- Jordan Borean (@jborean93)
+'''
+
+EXAMPLES = '''
+- name: reset pulp content
+  reset_pulp:
+    pulp_api: http://galaxy:24817
+    galaxy_ng_server: http://galaxy/api/galaxy/
+    url_username: username
+    url_password: password
+    repository: published
+    namespaces:
+    - namespace1
+    - namespace2
+'''
+
+RETURN = '''
+#
+'''
+
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.common.text.converters import to_text
+
+
+def invoke_api(module, url, method='GET', data=None, status_codes=None):
+    status_codes = status_codes or [200]
+    headers = {}
+    if data:
+        headers['Content-Type'] = 'application/json'
+        data = json.dumps(data)
+
+    resp, info = fetch_url(module, url, method=method, data=data, headers=headers)
+    if info['status'] not in status_codes:
+        module.fail_json(url=url, **info)
```

Comment:
This sometimes crashes with
```python-traceback
Traceback (most recent call last):
  File "<stdin>", line 121, in <module>
  File "<stdin>", line 113, in _ansiballz_main
  File "<stdin>", line 61, in invoke_module
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 209, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_reset_pulp_payload_8mi3ul8x/ansible_reset_pulp_payload.zip/ansible/modules/reset_pulp.py", line 211, in <module>
  File "/tmp/ansible_reset_pulp_payload_8mi3ul8x/ansible_reset_pulp_payload.zip/ansible/modules/reset_pulp.py", line 197, in main
  File "/tmp/ansible_reset_pulp_payload_8mi3ul8x/ansible_reset_pulp_payload.zip/ansible/modules/reset_pulp.py", line 197, in <listcomp>
  File "/tmp/ansible_reset_pulp_payload_8mi3ul8x/ansible_reset_pulp_payload.zip/ansible/modules/reset_pulp.py", line 102, in delete_pulp_distribution
  File "/tmp/ansible_reset_pulp_payload_8mi3ul8x/ansible_reset_pulp_payload.zip/ansible/modules/reset_pulp.py", line 87, in invoke_api
TypeError: ansible.module_utils.basic.AnsibleModule.fail_json() got multiple values for keyword argument 'url'
```

@jborean93 does Pulp API always returns `url` in their responses?